include/albatross/src/indexing/group_by.hpp

```diff
@@ -174,12 +190,40 @@ template <typename KeyType, typename ValueType> class GroupedBase {
   std::map<KeyType, ValueType> map_;
 };
 
-template <typename KeyType, typename ValueType>
+template <typename KeyType, typename ValueType, typename Enable>
 class Grouped : public GroupedBase<KeyType, ValueType> {
   using Base = GroupedBase<KeyType, ValueType>;
   using Base::Base;
 };
 
+/*
+ * Some specialized functionality for arithmetic value types.
+ */
+template <typename KeyType, typename ValueType>
+class Grouped<KeyType, ValueType,
+              std::enable_if_t<std::is_arithmetic<ValueType>::value>>
+    : public GroupedBase<KeyType, ValueType> {
+  using Base = GroupedBase<KeyType, ValueType>;
+  using Base::Base;
+
+public:
+  auto sum() const {
+    ValueType output = 0.;
+    for (const auto &pair : this->map_) {
+      output += pair.second;
+    }
+    return output;
+  }
+
+  auto mean() const {
+    double output = 0.;
```

Comment:
I guess for this function it's a more nuanced problem and we wouldn't really want to use `ValueType` here. Or at least we'd want to calculate the sum differently in the case of it being an integer...

I guess a `double` is the best and easiest choice.